frontend/src/routes/UserManagement/Identities/Users/UserYaml.tsx

```diff
@@ -2,19 +2,61 @@
 import { PageSection } from '@patternfly/react-core'
 import { useParams } from 'react-router-dom-v5-compat'
 import { useTranslation } from '../../../../lib/acm-i18next'
-import { AcmPage, AcmPageContent, AcmPageHeader } from '../../../../ui-components'
+import { useQuery } from '../../../../lib/useQuery'
+import { listUsers } from '../../../../resources/rbac'
+import { useMemo, useState, useEffect } from 'react'
+import { dump } from 'js-yaml'
+import YamlEditor from '../../../../components/YamlEditor'
+import { AcmLoadingPage } from '../../../../ui-components'
 
 const UserYaml = () => {
   const { t } = useTranslation()
   const { id = undefined } = useParams()
+  const { data: users, loading } = useQuery(listUsers)
+  const [editorHeight, setEditorHeight] = useState<number>(500)
 
-  return (
-    <AcmPage header={<AcmPageHeader title={t('User YAML')} description={`User: ${id}`} />}>
-      <AcmPageContent id="user-yaml">
-        <PageSection>User YAML page for ID: {id}</PageSection>
-      </AcmPageContent>
-    </AcmPage>
-  )
+  const user = useMemo(() => {
+    if (!users || !id) return undefined
+    return users.find((u) => u.metadata.uid === id || u.metadata.name === id)
+  }, [users, id])
+
+  useEffect(() => {
+    function handleResize() {
+      let editorHeight = window.innerHeight - 260
+      const globalHeader = document.getElementsByClassName('co-global-notification')
+      if (globalHeader.length > 0) {
+        editorHeight = editorHeight - globalHeader.length * 33
+      }
+
+      setEditorHeight(editorHeight)
+    }
+    handleResize()
+
+    window.addEventListener('resize', handleResize)
+
+    return () => window.removeEventListener('resize', handleResize)
+  }, [])
```

Comment:
wouldn't you move this to external custom hook and use that from here and then refactor frontend/src/routes/Governance/policies/policy-details/PolicyTemplateYaml.tsx to take it too, wdyt?
I'm thinking what it is about to come for either roles, groups, service accounts... for sure quality checks will complain about code duplication